module/zfs/dmu_traverse.c

```diff
@@ -241,6 +235,12 @@ traverse_visitbp(traverse_data_t *td, const dnode_phys_t *dnp,
 		ASSERT(0);
 	}
 
+	if (td->td_resume != NULL &&
+	    memcmp(zb, td->td_resume, sizeof (*zb)) == 0) {
+		/* found the resume point; no longer resuming */
+		memset(td->td_resume, 0, sizeof (*zb));
+	}
```

Comment:
It looks to me that in case of td_resume match if TRAVERSE_POST is set resume_skip_check() will return RESUME_SKIP_CHILDREN and the `goto post;` in the switch above will jump around this code, so resume point will never be cleared.  Am I missing something?

Reply:
I think you're right, and this works because we no longer rely on td_resume being zeroed (except perhaps as a very minor performance improvement?).  I'll see about simplifying this code to make it more clear.